test/NuGet.Core.Tests/NuGet.Protocol.Tests/HttpSource/HttpRetryHandlerTests.cs

```diff
@@ -208,6 +213,8 @@ public async Task HttpRetryHandler_MultipleTriesTimed()
             TestEnvironmentVariableReader testEnvironmentVariableReader = GetEnhancedHttpRetryEnvironmentVariables();
             Func<HttpRequestMessage, HttpResponseMessage> handler = requestMessage =>
             {
+                Thread.Sleep(TimeSpan.FromMilliseconds(10));
```

Comment:
If this is resolving it, then it's extremely likely we have a precision problem. 

What OS does it fail on?

Reply:
The logging shows that the call completed in 0ms which seems odd but technically we expect HTTP calls to take some amount of time.  The test is expecting the overall wall clock time to be at least 500ms which indicates that it was retried 5 times.  Adding this delay just ensures that if the HTTP call was retried 5 times, it will take 500ms or more overall.  Ideally this call would return some information telling us exactly how many times it was retried but unfortunately the test is just relying solely on time which is not great...